src/txkube/testing/strategies.py

```diff
@@ -299,49 +299,47 @@ def deployments():
     )
 
 
-def deploymentlists():
+def _collections(cls, strategy, unique_by):
     """
-    Strategy to build ``DeploymentList``.
+    A helper for defining a strategy to build ``...List`` objects.
+
+    :param cls: The model class to instantiate.
+    :param strategy: A strategy to build elements for the collection.
+    :param unique_by: A key function compatible with the ``lists`` strategy.
     """
     return builds(
-        v1beta1.DeploymentList,
-        items=lists(
-            deployments(),
-            average_size=_QUICK_AVERAGE_SIZE,
-            max_size=_QUICK_MAX_SIZE,
-            unique_by=_unique_names_with_namespaces,
+        cls,
+        items=one_of(
+            none(),
+            lists(
+                strategy,
+                average_size=_QUICK_AVERAGE_SIZE,
+                max_size=_QUICK_MAX_SIZE,
+                unique_by=unique_by,
+            ),
         ),
     )
 
 
+def deploymentlists():
+    """
+    Strategy to build ``DeploymentList``.
+    """
+    return _collections(v1beta1.DeploymentList, deployments(), _unique_names_with_namespaces)
```

Comment:
Probably needs to stay this way.  It's not clear the version portion is actually meaningfully separable from the other part.  I haven't come across any types that exist in multiple versions (eg, v1beta1.Deployment + v1.Deployment).  Such a thing could exist, the swagger file is 41kloc after all, and I haven't read it all...

But for now I think it makes sense to treat "v1beta1.DeploymentList` as just one thing, not really one particular version of one thing.